standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -7109,22 +7121,37 @@ protected String describeResult() {
   }
 
   @Override
-  public List<HiveObjectPrivilege> listPrincipalDBGrantsAll(
-      String principalName, PrincipalType principalType) {
-    try (QueryWrapper queryWrapper = new QueryWrapper()) {
-      return convertDB(listPrincipalAllDBGrant(principalName, principalType, queryWrapper));
+  public List<HiveObjectPrivilege> listPrincipalDBGrantsAll(String principalName, PrincipalType principalType) {
+    List<HiveObjectPrivilege> results = Collections.emptyList();
+    try {
+      openTransaction();
+      results = convertDB(listPrincipalAllDBGrant(principalName, principalType));
+      commitTransaction();
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    } finally {
+      rollbackAndCleanup(true, null);
```

Comment:
The transaction may unable to rollback when exception happens.

Reply:
Why is that?

Reply:
https://github.com/apache/hive/blob/871ee8009380e1bab160b58dc378a7f668c64584/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java#L11729-L11739
I'm wondering that the opened transaction may be unable to be committed when exception happens on ```listPrincipalAllDBGrant```  or rolled back as we set the ```success``` to true  on the method ```rollbackAndCleanup```. Could you explain more about why set ```success``` to true here?  Thanks @belugabehr

Reply:
@dengzhhu653 

This particular method is a top-level method and it is read-only.  Therefore, there will never be anything to roll back.

Reply:
When exception happens in this method, if the same ObjectStore tries to create table afterwards,  he may be unable to commit the transaction as ```openTrasactionCalls``` should be greater than 0 or rollback as the method ```commitTransaction``` returns true, the modification would be lost when metastore client shutdown.

Reply:
@dengzhhu653 OK, I am following you now.  I will address this is new push.  Thanks so much.

This whole setup is pretty wonky and I'd like to address, but in another JIRA.

Reply:
I found that using ```try (QueryWrapper wrapper = new QueryWrapper())``` does not need to add a catch clause comparing to ```try(Query q = pm.newQuery(....))```.